agentops/llms/__init__.py

```diff
@@ -0,0 +1,127 @@
+                        provider.override()
+                    else:
+                        logger.warning(
+                            f"Only LiteLLM>=1.3.1 supported. v{module_version} found."
+                        )
+                    return  # If using an abstraction like litellm, do not patch the underlying LLM APIs
+
+                if api == "openai":
+                    # Patch openai v1.0.0+ methods
+                    if hasattr(module, "__version__"):
+                        module_version = parse(module.__version__)
+                        if module_version >= parse("1.0.0"):
+                            provider = OpenAiProvider(self.client)
+                            provider.override()
+                        else:
+                            raise DeprecationWarning(
+                                "OpenAI versions < 0.1 are no longer supported by AgentOps. Please upgrade OpenAI or "
+                                "downgrade AgentOps."
+                            )
+
+                if api == "cohere":
+                    # Patch cohere v5.4.0+ methods
+                    module_version = version(api)
+                    if module_version is None:
+                        logger.warning(
+                            f"Cannot determine Cohere version. Only Cohere>=5.4.0 supported."
+                        )
+
+                    if Version(module_version) >= parse("5.4.0"):
+                        provider = CohereProvider(self.client)
+                        provider.override()
+                    else:
+                        logger.warning(
+                            f"Only Cohere>=5.4.0 supported. v{module_version} found."
+                        )
+
+                if api == "ollama":
+                    module_version = version(api)
+
+                    if Version(module_version) >= parse("0.0.1"):
+                        provider = OllamaProvider(self.client)
+                        provider.override()
+                    else:
+                        logger.warning(
+                            f"Only Ollama>=0.0.1 supported. v{module_version} found."
+                        )
+
+                if api == "groq":
+                    module_version = version(api)
+
+                    if Version(module_version) >= parse("0.9.0"):
+                        provider = GroqProvider(self.client)
+                        provider.override()
+                    else:
+                        logger.warning(
+                            f"Only Groq>=0.9.0 supported. v{module_version} found."
+                        )
+
+    def stop_instrumenting(self):
+        openai_provider = OpenAiProvider(self.client)
+        openai_provider.undo_override()
+
+        groq_provider = GroqProvider(self.client)
+        groq_provider.undo_override()
+
+        cohere_provider = CohereProvider(self.client)
+        cohere_provider.undo_override()
```

Comment:
What about litellm and ollama?

Reply:
Okay, ill add these and change the manual tests to utilize it